README.md

```diff
@@ -1 +1,78 @@
 # rancher-desktop-networking
+`rancher-desktop-networking` is a simple layer 2 switch that allows connectivity between the VM and the host over `AF_VSOCK` written in Go using [gvisor's](https://github.com/google/gvisor) network stack and [gvisor-tap-vsock](https://github.com/google/gvisor). In addition to piping the ethernet frames, it also provides a DNS, DHCP services and dynamic port forwarding.
+
+Rancher Desktop Networking consists of three main components: host-switch, vm-switch, and network-setup. Below is a brief explanation of what each component does.
+
+## How it works
+```mermaid
+flowchart  LR;
+ subgraph Host["HOST"]
+ subgraph hostSwitch["host-switch.exe"]
+ vsockHost{"main loop"}
+ eth(("reconstruct ETH frames"))
+ syscall(("OS syscall"))
+ dhcp["DHCP"]
+ dns["DNS"]
+ api["API"]
+portForwarding["Port Forwarding"]
+ vsockHost <----> eth
+ eth <----> syscall
+ vsockHost ----> dhcp
+ vsockHost ----> dns
+ vsockHost ----> api
+ vsockHost ----> portForwarding
+ end
+ end
+ subgraph VM["VM"]
+ subgraph netNs["Isolated Network Namespace"]
+ tapDevice("eth0")
+ subgraph vmSwitch["vm-switch"]
+ vsockVM{"VM Switch"}
+ ethVM(("listens for ETH frames\n from TAP Device"))
+ ethVM <----> vsockVM
+ end
+ tapDevice <----> ethVM
+ init["/sbin/init"]
+ end
+ network-setup["/usr/local/bin/network-setup"]
```

Comment:
This (and `init`) is just lying there doing nothing. I guess that's showing its relation with the network namespace, so that's kind of okay?